src/common/procedure/src/local.rs

```diff
@@ -440,17 +473,38 @@ impl ProcedureManager for LocalManager {
     }
 
     fn start(&self) -> Result<()> {
-        self.remove_outdated_meta_task
-            .start(common_runtime::bg_runtime())
-            .context(StartRemoveOutdatedMetaTaskSnafu)?;
+        // The previous value should be false
+        if !self.running.swap(true, Ordering::Relaxed) {
+            let mut state = self.state.lock().unwrap();
```

Comment:
Maybe we can use the lock of `self.state` to protect `start()`/`stop()`? We can keep the running flag and background tasks consistent and then document this in the comment.
```rust
let mut state = self.state.lock().unwrap();
if state.is_none() {
    // Already started
    return;
}
// We can assert running here.

// Starts task
// Set running.
```